.github/workflows/gradle-build-pr.yml

```diff
@@ -96,16 +103,50 @@ jobs:
           # export the original IP list so it can be restored during cleanup
           RESTORE_IPS=$(printf '%s\n' $CUR_IPS | jq -s -r 'add | unique | join(",")')
           echo "::set-env name=RESTORE_IPS::${RESTORE_IPS}"
-      # build and push container to integration
-      - name: "Set up integration environment"
+      - name: Check for available namespace to run tests in
+        run: |
+          NAMESPACES=("integration-1" "integration-2" "integration-3" "integration-4" "integration-5")
+          find_namespace ()
+          {
+            for i in "${NAMESPACES[@]}"
+            do
+              if kubectl get secrets -n ${i} ${i}-inuse > /dev/null 2>&1; then
+                printf "Namespace ${i} in use Skipping\n"
+              else
+                printf "Namespace ${i} not in use Deploying integration test to ${i}\n"
+                kubectl create secret generic ${i}-inuse --from-literal=inuse=${i} -n ${i}
+                tail=$(echo $i | awk -F- {'print $2'})
+                echo "::set-env name=IT_JADE_API_URL::https://jade-${tail}.datarepo-integration.broadinstitute.org"
+                echo "::set-env name=NAMESPACEINUSE::${i}"
+                return 0
+              fi
+            done
+            sleep 60
+            find_namespace
+          }
+          find_namespace
+      # Build the Docker image
+      - name: Build new api container and push to gcr
         env:
           DEV_PROJECT: broad-jade-dev
-          INTEGRATION_API: integration-jade-datarepo-api
         run: |
+          GCR_TAG=$(git rev-parse --short HEAD)
+          echo "::set-env name=GCR_TAG::${GCR_TAG}"
           ./gradlew jib
-          kubectl --namespace integration --record deployment/${INTEGRATION_API} \
-            set image deployment/${INTEGRATION_API} \
-            "${INTEGRATION_API}=gcr.io/${DEV_PROJECT}/jade-data-repo:$(git rev-parse --short HEAD)"
+      - name: Deploying new image to kubernetes
+        env:
+          DEV_PROJECT: broad-jade-dev
+          SECRET_CHART_VERSION: 0.0.4
+          DEPLOY_CHART_VERSION: 0.0.7
+        run: |
+          helm namespace upgrade ${NAMESPACEINUSE}-secrets datarepo-helm/create-secret-manager-secret --version=${SECRET_CHART_VERSION} \
+            --install --namespace ${NAMESPACEINUSE} -f \
+            "https://raw.githubusercontent.com/broadinstitute/datarepo-helm-definitions/master/integration/${NAMESPACEINUSE}/${NAMESPACEINUSE}Secrets.yaml"
+          helm namespace upgrade ${NAMESPACEINUSE}-jade datarepo-helm/datarepo --version=${DEPLOY_CHART_VERSION} --install \
+            --namespace ${NAMESPACEINUSE} -f \
+            "https://raw.githubusercontent.com/broadinstitute/datarepo-helm-definitions/master/integration/${NAMESPACEINUSE}/${NAMESPACEINUSE}Deployment.yaml" \
+            --set "datarepo-api.image.tag=${GCR_TAG}"
+      # build and push container to integration
```

Comment:
Also, I think this comment applies more to the "deploying new image to kubernetes" section

Reply:
deleted